src/currency-field/CurrencyField.ts

```diff
@@ -232,220 +292,223 @@ export class CurrencyField extends OmniFormElement {
-                    this._parseAmount(
-                        input.value.substring(0, caretPosition!) +
-                            input.value.substring(caretPosition! + diff, input.value.indexOf(this.fractionalSeparator))
-                    ) as number
-                ).then((res) => {
-                    input.value = res + this.fractionalSeparator + fractionPart;
-                });
+            // Check if the content being pasted is at the end of the input.
+            else if (input.selectionStart === input.value.length) {
+                centValue = this._convertToCents(input.value + numericPastedData);
             } else {
-                await this._formatToCurrency(
-                    input.value.substring(0, caretPosition!) + input.value.substring(caretPosition! + diff, input.value.length + 1)
-                ).then((res) => {
-                    input.value = res;
-                });
+                centValue = this._convertToCents(
+                    input.value.slice(0, input.selectionStart as number) + numericPastedData + input.value.slice(input.selectionEnd as number)
+                );
             }
 
-            e.preventDefault();
-            this.value = this._formatToFloat(input.value);
+            // Extract the amount part of the cent value.
+            const amountPart = centValue.substring(0, centValue.length - this.fractionalPrecision);
+            // Extract the cents part of the cent value.
+            const fractionPart = centValue.slice(-this.fractionalPrecision);
 
-            await this.updateComplete;
+            const parsedAmountPart = this._parseAmount(amountPart);
+            input.value = parsedAmountPart + this.fractionalSeparator + fractionPart;
 
-            this._inputElement?.setSelectionRange(caretPosition, caretPosition);
-        }
-
-        // Delete fraction (cents) part if delete button is clicked.
-        if (input.value.charAt(caretPosition!) === this.fractionalSeparator && e.key.toLowerCase() === 'delete') {
-            input.value = input.value.substring(0, input.value.indexOf(this.fractionalSeparator));
-            this.value = this._formatToFloat(input.value);
-            return;
-        }
+            const floatValue = this._formatToFloat(this._inputElement?.value as string);
+            this.value = floatValue;
+            this._dispatchCustomEvent(this.value as number);
 
-        // Copy currency field selection to clipboard.
-        if (e.ctrlKey && e.key.toLowerCase() === 'c') {
-            navigator.clipboard.writeText(input.value);
             return;
-        }
-
-        // Stop alpha keys from moving the caret position.
-        if (e.key >= 'a' && e.key <= 'z') {
+        } else {
+            // If pasted value is not valid position the caret to the end of the input.
             e.preventDefault();
+            setTimeout(() => {
+                input.selectionStart = input.selectionEnd = 10000;
```

Comment:
Should this not be based on the input value length instead of a harcoded value? What if the input has more than 10000 characters?
```suggestion
                input.selectionStart = input.selectionEnd = input.value?.length ?? 0;
```

Reply:
Implemented the suggested implementation